adminlib.php

```diff
@@ -181,7 +181,7 @@ private function view_plugins_table() {
         $table->setup();
 
         $plugins = $this->get_sorted_plugins_list();
-        $shortsubtype = substr($this->subtype, strlen('surveypro'));
+        $shortsubtype = \core_text::substr($this->subtype, strlen('surveypro'));
```

Comment:
This looks really strange... if we use `core_text` for `substr` we should also be using `core_text` for `strlen`.

I mean, what's the point about getting correct UTF-8 substring... if the calculation of the length is not UTF-8. Both need to be the same.

```suggestion
        $shortsubtype = \core_text::substr($this->subtype, \core_text::strlen('surveypro'));
```

Another small detail... it's not really needed to use the backslash (\) when you are in global-scope files, like this one. Only when you're in a file that belongs to any other namespace is when the backslash is required.

In any case, this detail is only a matter of preference, so if you want to use them... all right.

Reply:
You are right, as usual. I agree with you.
Anyway, it is fantastic the detail you can discover.
I COMPLETELY missed the strlen in line 184.

THANK YOU!